lib/spack/spack/build_environment.py

```diff
@@ -1034,7 +1040,7 @@ def get_cmake_prefix_path(pkg):
     ordered_build_link_deps = spack_built + externals
     build_link_prefixes = filter_system_paths(
         x.prefix for x in ordered_build_link_deps)
-    return build_link_prefixes
+    return build_link_prefixes + extra_prefix_paths 
```

Comment:
I'm concerned that this isn't following the order promoted by the comment, but I don't immediately have any good suggestions for how to restore that (in a way that doesn't duplicate code - I think it could be accomplished with some refactoring). I won't have time to make suggestions about that until Wednesday.

Also either way, if this were taken as-is, the comment at the beginning of the method is no-longer accurate. I wrote that (and this) so I would also want to look into why I avoided doing that in the first place.

Reply:
@scheibelp I figured this proposed change would take some time to figure out: I'm assuming we stopped considering this variable for a reason (given the comment), but I looked at the commit it came from and it wasn't obvious to me.

I'll clean up the style issue tomorrow, and make the comment make sense at the very least.

Reply:
To address my original concern:

> Also either way, if this were taken as-is, the comment at the beginning of the method is no-longer accurate. I wrote that (and this) so I would also want to look into why I avoided doing that in the first place.

It seems like the original implementation of `get_cmake_prefix_path` also had this feature when I wrote https://github.com/spack/spack/commit/916cdfbb56b2abfc380ce1b4cac0d6494b7f2628, so I think that comment was just saying I was also deciding not to handle it here.